### PR TITLE
TEIID-5764 defaulting to v2 and adding version parameter handling

### PR DESCRIPTION
**Pull request by shawkins**

### CODE REVIEW

File: olingo/src/main/java/org/teiid/olingo/web/OpenApiHandler.java [hunk 1]
```diff
@@ -50,17 +50,18 @@
...
 public class OpenApiHandler {
 
-    enum OpenApiVersion {
-        V2("2.0", "/swagger.json"), //$NON-NLS-1$ //$NON-NLS-2$
-        V3("3.0", "/openapi.json"); //$NON-NLS-1$ //$NON-NLS-2$
+    private static final String SWAGGER_JSON = "/swagger.json"; //$NON-NLS-1$
+    private static final String OPENAPI_JSON = "/openapi.json"; //$NON-NLS-1$
 
-        private OpenApiVersion(String key, String uri) {
+    public enum OpenApiVersion {
```
Comment (rareddy) [inline olingo/src/main/java/org/teiid/olingo/web/OpenApiHandler.java]: Are you sure you want to introduce our own versioning mechanism over just using OpenApi spec versioning scheme (major.minor.macro) and gleaning the version info from it?

Comment (shawkins) [reply olingo/src/main/java/org/teiid/olingo/web/OpenApiHandler.java]: You mean am I worried about 2/3 vs. 2.0.0 and 3.0.0, no.  If/when minor/micro are utilized we can evaluate how we want this to be handled - at worst we'd need version parsing and select the specification that is closest, etc.  So 2/3 seems good for now.

**Comment by rareddy** [comment]
Can you also merge this 12.2.1 ?

**Comment by shawkins** [comment]
We're all set.

**Pull request merged**